kernel/src/debug.rs

```diff
@@ -80,14 +80,14 @@ const BUF_SIZE: usize = 1024;
 pub struct DebugWriter {
     driver: Option<&'static Driver>,
     pub grant: Option<*mut u8>,
-    output_buffer: [u8; BUF_SIZE],
-    output_head: usize,
-    output_tail: usize,
+    pub output_buffer: [u8; BUF_SIZE],
```

Comment:
Is making these pub the right call? Or should debug just gain a new public function called "flush" that runs the code, and then on panic, flush gets called?

Reply:
That's not so easy, as the flush operation is using the `writer`, which is a mutable reference to `WRITER` which is an instance of the `struct Writer` defined in `io.rs`, so there's not an immediately obviously clean way to pass the writer to the debug module.

I have designs on generalizing and cleaning up the panic routines more in the long run, which could address this. In the short term, I think it's fine to just make these public.